src/main/java/factionsplusplus/models/Group.java

```diff
@@ -97,47 +162,67 @@ public String getMemberListSeparatedByCommas() {
         return "";
     }
 
-    public boolean addOfficer(UUID newOfficer) {
-        if (!officers.contains(newOfficer)) {
-            officers.add(newOfficer);
+    public boolean addOfficer(UUID playerUuid) {
+        if (! this.isMember(playerUuid)) this.addMember(playerUuid);
+
+        GroupMember member = this.getMember(playerUuid);
+        if (! member.hasRole(GroupRole.Officer)) {
+            member.addRole(GroupRole.Officer);
             return true;
         }
+
         return false;
     }
 
-    public boolean removeOfficer(UUID officerToRemove) {
-        return officers.remove(officerToRemove);
+    public boolean removeOfficer(UUID playerUuid) {
+        if (! this.isMember(playerUuid)) return false;
+
+        GroupMember member = this.getMember(playerUuid);
+        if (member.hasRole(GroupRole.Officer)) {
+            member.removeRole(GroupRole.Officer);
+            return true;
+        }
+
+        return false;
     }
 
-    public boolean isOfficer(UUID uuid) {
-        return officers.contains(uuid);
+    public boolean isOfficer(UUID playerUuid) {
+        if (! this.isMember(playerUuid)) return false;
+
+        return this.getMember(playerUuid).hasRole(GroupRole.Officer);
     }
 
     public int getNumOfficers() {
-        return officers.size();
+        return (int) this.getMembers()
+            .stream()
+            .filter(member -> member.hasRole(GroupRole.Officer))
+            .count();
     }
 
-    public List<UUID> getOfficerList() {
-        return officers;
+    public List<GroupMember> getOfficerList() {
+        return this.getMembers()
+            .stream()
+            .filter(member -> member.hasRole(GroupRole.Officer))
+            .toList();
     }
 
     public int getPopulation() {
-        return members.size();
+        return this.members.size();
     }
 
-    public void invite(UUID playerName) {
-        Player player = getServer().getPlayer(playerName);
+    public void invite(UUID playerUuid) {
+        Player player = getServer().getPlayer(playerUuid);
         if (player != null) {
-            UUID playerUUID = getServer().getPlayer(playerName).getUniqueId();
-            invited.add(playerUUID);
+            playerUuid = getServer().getPlayer(playerUuid).getUniqueId();
```

Comment:
we already are given the UUID in the method call, no need to ask the server for it right?